### PR TITLE
Lpf/release doc update

### PR DESCRIPTION
**Pull request by laurenfrederick**
**Summary:** Summary of changes

Addresses [CUMULUS-XX: Develop amazing new feature](https://bugs.earthdata.nasa.gov/browse/CUMULUS-XXX)

## Changes

* Detailed list or prose of changes
* ...

## PR Checklist

- [ ] Update CHANGELOG
- [ ] Unit tests
- [ ] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

### CODE REVIEW

File: docs/development/release.md [hunk 1]
```diff
@@ -260,7 +260,9 @@ Finally, you need to reproduce the version update changes back to master.
 
 If this is the latest version, you can simply create a PR to merge the minor version base branch back to master.
 
-Do not merge `master` back into the release branch **until**:
+Do not merge `master` back into the release branch since we want the release branch to _just_ have the code from the release.  Instead, create a new branch off of the release branch and merge that to master. You can freely merge master into this branch and delete it when it is merged to master.
+
+Do not merge `master` back into the release merge branch **until**:
```
Comment (Jkovarik) [inline docs/development/release.md:265]: Is this and L 267-271 needed now, given we're not bringing master changes direct to the release branch in favor of the merge branch?

**Review by Jkovarik — CHANGES_REQUESTED** [review]
Looks good, just one minor question/issue to address 👍🏻

File: docs/development/release.md (hunk 1 above)
Comment (laurenfrederick) [reply docs/development/release.md]: I took them out, then added them back in. Then I read them again and took them out.

**Review by Jkovarik — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**